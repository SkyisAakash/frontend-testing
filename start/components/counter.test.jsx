import React from 'react';
import { shallow } from 'enzyme'
import Counter from './counter'
import Foo from './foo'
describe("counter", () => {
    it("starts with 0", () => {
        let container  = shallow(<Counter />);
        let counterCount = container.state().count;
        expect(counterCount).toBe(0);
    })

    it("renders foos", () => {
        let container = shallow(<Counter />);
        expect(container.find("Foo").length).toBe(0);
        container.find("button.add1").simulate("click");
        expect(container.find("Foo").length).toBe(1);
        container.find("button.add1").simulate("click");
        container.find("button.add1").simulate("click");
        container.find("button.add1").simulate("click");
        expect(container.find("Foo").length).toBe(4);
    })
})