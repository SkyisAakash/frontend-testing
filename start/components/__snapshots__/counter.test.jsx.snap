// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`counter matches snapshot 1`] = `
<div>
  <p>
    count: 
    0
     
  </p>
  <button
    className="add1"
    onClick={[Function]}
  />
</div>
`;
